File: OOP_Principles/Polymorphism/Polymorphism.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace c_sharp_design_patterns.OOP_Principles.Polymorphism
{
    public class Polymorphism
    {
        /*
         * Polymorphism is the ability of an object to take many forms.
         */

        public void Example()
        {
            List<Vehicle> vehicles = new List<Vehicle>();
            vehicles.Add(new Car()); // Honk is base implementation
            vehicles.Add(new Bike());
            vehicles.Add(new Boat());

            foreach (Vehicle vehicle in vehicles)
            {
                vehicle.Honk();
            }
        }
    }
}

File: OOP_Principles/Polymorphism/Polymorphism.java
package OOP_Principles.Polymorphism;

public class Polymorphism {

  /*
   * Polymorphism is the ability of an object to take many forms.
   */

  public void example() {
    Vehicle[] vehicles = { new Car(), new Bike(), new Boat() };

    for (Vehicle vehicle : vehicles) {
      vehicle.Honk();
    }
  }
}
